src/charm.py

```diff
@@ -197,6 +220,31 @@ def _restart(self, event: EventBase):
 
         self._on_start(event=event)
 
+    def _on_certificate_relation_joined(self, event):
+        self.tls_certificates.request_certificate(
+            cert_type="server",
+            common_name=f"{os.uname()[1]}",
+        )
+
+    def _on_certificate_available(self, event):
+        certificate_data = event.certificate_data
+        path = Path(TLS_STORE_DIR)
+        path.mkdir(parents=True, exist_ok=True)
+        for k, v in certificate_data.items():
+            f = open(f"{TLS_STORE_DIR}/{k}", "w")
+            f.write(v)
+            f.close()
+        self.snap.write_properties(
+            properties=self.zookeeper_config.create_properties(self.config, self.cluster),
+            property_label="zookeeper",
+        )
+        self.on[self.restart.name].acquire_lock.emit()
+
+    def _on_certificate_relation_departed(self, event: EventBase):
+        for file in ["ca", "cert", "key", "common_name"]:
+            if os.path.exists(f"{TLS_STORE_DIR}/{file}"):
+                os.remove(f"{TLS_STORE_DIR}/{file}")
```

Comment:
Should there be a check if config `retain-certs` is True?

Reply:
I think I Ctrl+Z'd a few too many times on this one.  I'll add the check back.